### ISSUE TITLE
Greedy solver not chosen despite ILP `cbc` taking >6min and thus >10s

### ISSUE THREAD

**Issue by corneliusroemer** [snakemake/snakemake#1374]
**Snakemake version**
6.7.0

**Describe the bug**
Snakemake gets stuck for at least 5 minutes, not submitting any jobs. Inspection of what's going on via `htop` showed 100% CPU for a command called `cbc` which seems to be a solver.

![image](https://user-images.githubusercontent.com/25161793/152471602-172df289-80c7-436a-aaae-5e6720214b8a.png)

![image](https://user-images.githubusercontent.com/25161793/152471615-3c1ca647-9023-4918-9387-71c75a4980a5.png)

The next step involved selecting which of 1000 jobs to run, but they are all equivalent.

When I hit `Ctrl+c` Snakemake seemed to recover and printed that it switched to _greedy_ solver:
```
Failed to solve scheduling problem with ILP solver in time (10s).
Falling back to greedy solver.
```

I read somewhere that usually that switch should happen after 10 seconds. In this case, that didn't happen.

The jobs were to be submitted on a SLURM cluster.

I'm not sure this is reproducible, but I thought I raise it anyways so that others who encounter similar issues can post below and know they're not alone.

**Workaround**
Hit `Ctrl+C` once

Hitting `Ctrl+c` manually works as a workaround
![image](https://user-images.githubusercontent.com/25161793/152471712-dd976d6a-7ef9-4de2-a791-901c8779bc89.png)

**Comment by dariober** [comment]
If it helps, there is a [--scheduler](https://snakemake.readthedocs.io/en/stable/executing/cli.html) option which defaults to "ilp" but can be switched to "greedy". Better than the CTRL-C trick?

**Comment by corneliusroemer** [comment]
Yes indeed, but had t tested. It's still a bug I think since the prompt explicitly says that it should switch after 10s automatically.

**Comment by mbhall88** [comment]
I am also seeing this. In my case, it took 12 hours to print this message - the pipeline has 45000 jobs.

**Comment by johanneskoester** [comment]
Fixed by #2938.
